common-libs/common/src/shared/seedTools.ts

```diff
@@ -128,12 +128,12 @@ export const convertDecryptedSeedBufferToString = (decryptedSeed: Buffer): strin
   return `${seedHex}++jolo`
 }
 
-export const processAnchoredElemDidSeed = (parsedSeed: ParseDecryptedSeedResult, did: string) => {
-  const { seed, externalKeys } = parsedSeed
+export const extendSeedWithDid = (decryptedFullSeed: string, did: string) => {
+  const { seed, metadata, externalKeys } = parseDecryptedSeed(decryptedFullSeed)
```

Comment:
In theory, `additionalData` can contain other things beyond `metadata` and `externalKeys`, and here you are throwing it away.

(This was already mentioned in the earlier comment)

Reply:
I don't like this approach, when it will contain smth we can easily modify this code. My position - it can be as is

Reply:
For me it looks like https://stackify.com/premature-optimization-evil/

Reply:
It is not a premature optimization (and premature optimization refers to _performance_ optimization).

You can write "add metadata.anchoredDid field to the existing object, no matter what else it contains".
Or you can write "extract externalKeys and metadata from the existing object, add anchoredDid field to metadata, combine externalKeys and metadata into a new object".

The former is simpler, does not require knowledge of how externalKeys interact with additionalData, and future-proof.
The latter requires this function to know about externalKeys, and can cause subtle bugs in the future when it will erase everything else that `additionalData` can hold by then.

Reply:
I mean, it is just time-wasting as and premature optimization - not more. How smth can be changed to additionalData? Only one way - open this file and add some changes, then add tests for that. 
https://levelup.gitconnected.com/the-problem-you-solve-is-more-important-than-the-code-you-write-d0e5493132c6

Reply:
And the problem this function solving, as is clear from its name, is "add anchored did to metadata". Not "add anchored did to metadata _and_ remove everything that's not external keys or metadata from additional data".

Reply:
There are nothing that's not external keys or metadata in additional data

Reply:
At this moment, yes. But `extendSeedWithDid` does not have to know about that.